### ISSUE TITLE
[Feature] Matrix multiplication (matmul) operation

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#48]
## Summary

Add `MatMulOp` for matrix multiplication (dot product of 2D+ tensors).

## Motivation

Matrix multiplication is fundamental for linear transformations, attention score computation, and projection operations. While dart_tensor_preprocessing is focused on preprocessing, matmul is needed for:
- Linear whitening transforms (`torchvision.transforms.LinearTransformation`)
- Custom projection-based preprocessing
- Attention mask computation
- Post-processing linear layers

## Reference Implementations

- **PyTorch**: `torch.matmul(input, other)` — supports broadcasting for batch matmul — [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.matmul.html)
- **NumPy**: `numpy.matmul` / `@` operator
- **ONNX**: `MatMul` operator (opset 1+) — [ONNX spec](https://onnx.ai/onnx/operators/onnx__MatMul.html)
- **dart_tensor**: `dot()` method
- **ml_linalg**: Matrix multiplication with SIMD

## Verification

- ✅ ONNX `MatMul` operator exists (opset 1+)
- ✅ Both dart_tensor and ml_linalg provide matrix multiplication
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for matmul, dot, mm, matrix_multiply)

## Proposed API

```dart
// 2D matrix multiply: [M, K] × [K, N] → [M, N]
final result = MatMulOp(weight).apply(input);

// Batch matmul: [B, M, K] × [B, K, N] → [B, M, N]
final result = MatMulOp(weight).apply(batchInput);

// Convenience for common use cases
final linear = LinearTransformOp(weight: transformMatrix, bias: biasVector);
```

## Implementation Notes

- Support 2D×2D, 2D×1D (matrix-vector), and batched matmul
- Use tiled/blocked algorithm for cache efficiency (similar to existing resize blocking)
- SIMD-acceleratable using Float32x4 dot products
- Consider integration with `ml_linalg` for optimized SIMD matmul
- Add `OperationCapabilities`:
  - `onnxOpType: 'MatMul'`
  - `onnxOpsetVersion: 1`
  - `pytorchEquivalent: 'torch.matmul'`

## ROI Assessment

**Medium** — Not core to image preprocessing, but increasingly important for custom transforms and post-processing. The existing SIMD infrastructure can be leveraged for performance.
